hyperspy/_signals/sed.py

```diff
@@ -0,0 +1,465 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2016 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import division
+
+import traits.api as t
+import math
+import numpy as np
+import scipy.ndimage as ndi
+
+from hyperspy._signals.signal2d import Signal2D
+from hyperspy.decorators import only_interactive
+from hyperspy.gui.sed import SEDParametersUI
+from hyperspy.defaults_parser import preferences
+import hyperspy.gui.messages as messagesui
+
+
+class SEDPattern(Signal2D):
+    _signal_type = "SED_Pattern"
```

Comment:
Honestly I wasn't really sure what was best here -- yes it's certainly more general and I'd be happy with ElectronDiffraction, but the methods are only really ok for TEMElectronDiffraction spot patterns rather than say EBSD or Transmission Kikuchi... Not really sure what structure we should be going for thinking a few steps ahead.